torch/_inductor/lowering.py

```diff
@@ -4138,6 +4137,124 @@ def fn(idx):
     # TODO: should we force these to be realized?
     return rv
 
+def _adaptive_pooling_idx_max(kernel_maxes, in_sizes, out_sizes, return_index, loader):
```

Comment:
Did you end up implementing this copying `adaptive_pooling_avg`? If so, can you leave a note saying we should deduplicate this code once https://github.com/pytorch/pytorch/pull/116418 is merged.